source/speech/speech.py

```diff
@@ -2068,6 +2068,7 @@ def getControlFieldSpeech(  # noqa: C901
 			controlTypes.Role.LANDMARK,
 			controlTypes.Role.REGION,
 		)
+		and config.conf["documentFormatting"]["reportLandmarks"]

```

Comment:
This will disable reporting of groupings and propertyPages if reportLandmarks is false.
I would not add this line. But further down...

Reply:
A few lines down:
```
nameAndRole.extend(roleTextSequence)
```
Make this line conditional on the role being landmark or region, and reportLandmarks being true.
In other words, we want to speak the label for grouping, propertyPage, landmark or region what ever the users config. But, we only want to speak the role for landmarks and regions if reportLandmarks is true.